ams/server/src/main/java/com/netease/arctic/server/table/TableRuntime.java

```diff
@@ -180,12 +180,12 @@ public void setPendingInput(OptimizingEvaluator.PendingInput pendingInput) {
 
   public TableRuntime refresh(ArcticTable table) {
     return invokeConsisitency(() -> {
-      TableConfiguration configuration = tableConfiguration;
-      if (refreshSnapshots(table) || updateConfigInternal(table.properties())) {
+      boolean configChanged = updateConfigInternal(table.properties());
+      if (refreshSnapshots(table) || configChanged) {
         persistUpdatingRuntime();
       }
-      if (configuration != tableConfiguration) {
-        tableHandler.handleTableChanged(this, configuration);
+      if (configChanged) {
+        tableHandler.handleTableChanged(this, tableConfiguration);
```

Comment:
I think we should use the configuration before changed, because the interface is 
```
public interface TableRuntimeHandler {
  ...
  void handleTableChanged(TableRuntime tableRuntime, OptimizingStatus originalStatus);

  void handleTableChanged(TableRuntime tableRuntime, TableConfiguration originalConfig);
 ...
}

```

And the `originalConfig` means configuration before changed.

Reply:
You are right.